openapi-validator/src/main/java/io/ballerina/openapi/validator/ParameterValidator.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Override
+    public void validate() {
+        //Ballerina to OAS parameter validation , here we validate query, path parameters.
+        validateBallerina();
+
+        //OAS->Ballerina parameter validate
+        if (oasParameters != null) {
+            validateOpenAPI();
+        }
+    }
+
+    /**
+     * Validate OAS parameter against ballerina parameters.
+     */
+    @Override
+    public void validateOpenAPI() {
+
+        oasParameters.forEach(parameter -> {
+            if (parameter.get$ref() != null) {
+                Optional<String> parameterName = extractReferenceType(parameter.get$ref());
+                if (parameterName.isEmpty()) {
+                    return;
+                }
+                parameter = openAPI.getComponents().getParameters().get(parameterName.get());
+            }
+            if (!(parameter instanceof HeaderParameter) ||
+                    parameter.getIn() != null && !parameter.getIn().equals("header")) {
+                // headerValidation
+                AtomicBoolean isImplemented = new AtomicBoolean(false);
+                Parameter finalParameter1 = parameter;
+                parameters.forEach((paramName, paramNode) -> {
+                    // avoid headers
+                    if (finalParameter1.getName().equals(paramName)) {
+                        isImplemented.set(true);
+                    }
+                });
+                if (!isImplemented.get()) {
+                    // error message
+                    reportDiagnostic(context, CompilationError.MISSING_PARAMETER,
+                            location, severity, parameter.getName(), method, path);
+                }
+            }
+        });
+    }
+
+    /**
+     * This function is used to validate the ballerina resource parameter against to openapi parameters.
+     *
+     */
+    @Override
+    public void validateBallerina() {
+        for (Map.Entry<String, Node> parameter : parameters.entrySet()) {
+            boolean isExist = false;
+            String parameterName = unescapeIdentifier(parameter.getKey());
+            String ballerinaType;
+            //Todo: Nullable and default value assign scenarios
+            if (parameter.getValue() instanceof ParameterNode) {
+                ParameterNode paramNode = (ParameterNode) parameter.getValue();
+                if (paramNode instanceof RequiredParameterNode) {
+                    RequiredParameterNode requireParam = (RequiredParameterNode) paramNode;
+                    ballerinaType = requireParam.typeName().toString().trim().replaceAll("\\?", "");
+                } else {
+                    DefaultableParameterNode defaultParam = (DefaultableParameterNode) paramNode;
+                    ballerinaType = defaultParam.typeName().toString().trim().replaceAll("\\?", "");
+                }
+            } else {
+                ResourcePathParameterNode pathParameterNode = (ResourcePathParameterNode) parameter.getValue();
```

Comment:
Doing this type case within the else block might not be safe and future-proof. Better to use an `else if` condition to check the type before casting.